src/fs_clonefile.cpp

```diff
@@ -67,24 +67,24 @@ writen(const int     fd,
           return -1;
         }
 
-      nleft -= nwritten;
-      buf   += nwritten;
+      nleft -= (size_t)nwritten;
+      buf   += (size_t)nwritten;
     }
 
-  return count;
+  return nwritten;
```

Comment:
This function has nothing to do with FUSE. It's for copying data between to underlying drives.

A write, according to standards, does not return 0 unless it's asked to write 0. It can return from 1 to count or -1. That's why writen exists. To make sure that the whole buffer is written.

The problem is the outer function. It should exit on EOF but write should never return anything but 1 - bufsize or -1. And writen should manage being passed a zero bufsize.